### DISCUSSION TITLE
Building Bootstrap Dotnet for Gentoo (LibreSSL)

### DISCUSSION THREAD

**Discussion by blucybrb14de** [dotnet/source-build#4725] [Q&A]
**Hello,**

I'm currently attempting to port dotnet-sdk to Gentoo on a LibreSSL system, and attempting to create a new ebuild for dotnet-sdk. The trouble is, I am not able to build my own bootstrap when looking at the documentation, as it appears that the buildbootstrapcli.sh is missing, and not a part of a ./scripts directory on the main branch.

It seems to mention that it is possible build a bootstrap for dotnet-sdk for a LibreSSL system, using the build bootstrap, and I would like to attempt to give non-official support in the LibreSSL overlay, however the steps to do so [seemed to be obscured](https://github.com/dotnet/source-build/blob/b05c0daf8b26acd8f8862abe7fe5e0177234d5bb/Documentation/boostrap-new-os.md)

I did not try to look to look through the commit history, but I can't seem to find the relevant information in order to do so. If anyone could please provide any instructions to continue to bootstrap, I would love to continue to run and test LibreSSL on Gentoo, for the packages that depend on it, and to do so for the foreseeable future. 

Much Appreciated.

**Comment by mthalman** [comment]
Please follow this document instead: https://github.com/dotnet/source-build/blob/main/Documentation/bootstrapping-guidelines.md. The doc at https://github.com/dotnet/source-build/blob/main/Documentation/boostrap-new-os.md seems to be outdated or obsolete. I've logged https://github.com/dotnet/source-build/issues/4729 to get that addressed.

**Reply by blucybrb14de** [reply]
> Please follow this document instead: https://github.com/dotnet/source-build/blob/main/Documentation/bootstrapping-guidelines.md. The doc at https://github.com/dotnet/source-build/blob/main/Documentation/boostrap-new-os.md seems to be outdated or obsolete. I've logged #4729 to get that addressed.

Thanks for pointing that out. 

There seems to still be some things missing from the documentation, in order to build the bootstrap, I am using dotnet-sdk-8.0.404-linux.tar.gz, and I have extracted this to ~/dotnet-bootstrap, however these instructions seem to be confusing, as it just mentions modifying the files, in https://github.com/dotnet/runtime, runtime.json which is mentioned as an issue in the instructions. 

Just with the information provided, it is difficult for me to follow how that applies to building the bootstrap. However, with the previous instructions it seemed relatively straightforward to build the bootstrap using the cli, as I am still building a bootstrap for linux-x86_64, but using the libressl library instead. Which shares the same soname, and is mentioned as being possible to build.

If anyone is able to provide instructions or clarification that would be much appreciated.

**Reply by omajid** [reply]
We seem to have a documentation gap that describes how to actually bootstrap things. This was also called out by others in https://github.com/dotnet/source-build/discussions/4501.

The overall approach looks like this:

- Set up a system where an existing .NET SDK can run. Generally this would mean x86_64 with glibc with OpenSSL. This could be a VM.
- Set up a rootfs that provides a cross-compilation environment that can target your desired architecture/libc/crypto-libraries
- Run the build with specific commands and references to make it use that rootfs to cross-compile. My exact steps for another bootstrap scenario are documented here: https://github.com/dotnet/source-build/discussions/4501#discussioncomment-10501331
